### DISCUSSION TITLE
Some observations...

### DISCUSSION THREAD

**Discussion by PH5HP** [rejetto/hfs#414] [Ideas]
Hi Rejetto,

I was playing a little bit with the account possebilities. I created an account for myself where I can do everything. I requires a login.

1.
So if I click on my folder link than I am prompted for my login credentials. So I login and I am returning in the root folder of my fileserver. I would expect to see the contents of my personal folder (named 'henk' in this case) instead of the root folder of the fileserver, because I clicked on the  'henk' folder.

2.
I am in the 'Henk' folder. If I logout I get the login popup again. I would expect to return to the root of the fileserver instead.

Kind regards,
Henk - fileserver.ph5hp.nl

**Comment by rejetto** [comment]
1. i'm confused, i'm not sure i got exactly what you described, so we have to first see if I got you correctly.
You have the root configured with some content A, and a protected folder with some content B. You access the folder, then you click to navigate to the root, and you expect to see not A but B. Is that it?

2. you expect because you know the root is not protected, but that's true for your server, while not in the general case. Lucky you, it is easy to achieve by adding this in *Custom HTML > Before header*
```html
<script>
HFS.watchState('username', u => {
    if (!u) location = '/'
})
</script>
```
read it like "if username changes, and it's NOW empty, then redirect".

If I understand that this mechanism is interesting for many people, I can turn this into a plugin, or even an option of the program itself.

The only downside of this solution, is that using location to move around will cause the page to fully reload. Most people will not even care, but in next version I will include an alternative method to do it even better.

**Reply by PH5HP** [reply]
Hi,

About 1.  Forget that. I am wrong about that one. I first login and then I click on the protected folder because it is only visible after login. My mistake. Sorry.

About 2. You are right about that one. It is true I have an unprotected root and there will be people who have a protected root. I do not a solution to that one. The only thing I can come up with is that what happens if all folders are protected (root also). Do you get the login popup? Maybe you can check if there are unprotected folders in the root and if so then go to the root folder?

Kind regards,
Henk - PH5HP - fileserver.ph5hp.nl

**Comment by rejetto** [comment]
if and where to go on logout is a bit arbitrary, so, if anything, i would let the administrator decide.
Let me know if the solution I gave you is good.

**Reply by PH5HP** [reply]
Hi,

Yes, it does the trick. If I am in my personal map after login and I logout there then I am redirected to the root of my fileserver.

Kind regards,
Henk - fileserver.ph5hp.nl
